### ISSUE TITLE
Automated QA: Unit tests with other mocked infrastructure dependencies like logger, database, etc

### ISSUE THREAD

**Issue by pavlovcik** [ubiquity/ubiquibot#635]

**Comment by pavlovcik** [comment]
I'll make an exception for tests. I think it will be fast to merge in because we don't need to really QA test it.

**Comment by byteballet** [comment]
Since we migrated log from mezmo to supabase, is db test enough?

**Comment by pavlovcik** [comment]
Not sure what that means but sounds like @0xcodercrane territory

**Comment by web4er** [comment]
I am looking forward to this. 

Following are some points that I consider for high-quality unit tests.

- must be fast to run for programmers to get a quick feedback 
- should be deterministic
- unit tests must be isolated, one test should not depend on other
- test behavior, not implementation
- no heavy mocking. In a unit test, a unit is not necessarily a function or a class, it is a unit of requirements. 
- no forced high coverage by writing unnecessary tests.

**Comment by 0xcodercrane** [comment]
> Since we migrated log from mezmo to supabase, supabase test is enough, isn't it?

We've created an adapter for supabase so it's fine that have the unit tests for supabase adapter. no need for supabase-js itself

**Comment by 0xcodercrane** [comment]
> no heavy mocking. In a unit test, a unit is not necessarily a function or a class, it is a unit of requirements.

What do you mean by `unit of requirements`? In my opinion, it should be either a function or class.

**Comment by byteballet** [comment]
Wouldn't it be good to create another supbase db for testing?

**Comment by web4er** [comment]
> What do you mean by unit of requirements? In my opinion, it should be either a function or class.

You are right. I see not-so-good wording used by me here. Basically, what I am trying to say is if a function or a class that is subject under test, is calling another function, do not called mock that function. 

Developers try to achieve test in isolation. Here, they try to isolate function or class which is the subject under test. This results in heavy mocking. Test in isolation is for the "test" not the "subject under tests".

But you can mock for one of 2 reasons: 
- it slows down test speed, e.g. http request
- shared fixture where one test can affect another test e.g. file or db

**Comment by byteballet** [comment]
Currently in progress.

**Comment by byteballet** [comment]
> We've created an adapter for supabase so it's fine that have the unit tests for supabase adapter. no need for supabase-js itself

In case using supabase adapter, isn't the database affected?

**Comment by 0xcodercrane** [comment]
> > We've created an adapter for supabase so it's fine that have the unit tests for supabase adapter. no need for supabase-js itself
> 
> In case using supabase adapter, isn't the database affected? or should I just assume that the test is not done for production db?

The database isn't affected.
